### DISCUSSION TITLE
Configuration page error after disabling http api

### DISCUSSION THREAD

**Discussion by Barry40** [arendst/Tasmota#13505] [Q&A] [answered]
Hi, in tasmota 10.0 in sonoff pow r2, i cannot enter all configuration pages anymore since i disabled the http api.
Dunno what todo now since the firmware page is also gone.
It looks like only the index (home) page is working and the reboot, console, configuration and the rest gives error and will not open anymore?

**Comment by barbudor** [comment, marked as answer]
Did you used `WebServer 1` to set the web-server in User mode instead of Admin mode ?

Only solution is to send the command `WebServer 2` either by serial or by MQTT

**Reply by sfromis** [reply]
You could also do a "factory reset" of Tasmota to initial settings, if getting at it via MQTT/serial is too much trouble. Note, Tasmota has no API via http/rest, just a not optimized web server, slower and less reliable than MQTT.

**Reply by Barry40** [reply]
Okay, i will try!
Thanks. Will let you guys know.

**Reply by Barry40** [reply]
> Did you used `WebServer 1` to set the web-server in User mode instead of Admin mode ?
> 
> Only solution is to send the command `WebServer 2` either by serial or by MQTT

I do not use webserver 1 as far as i changed nothing in the "default" configuration.
mqtt / wi-fi / hostname and idx's, thats all..

**Reply by Barry40** [reply]
> Did you used `WebServer 1` to set the web-server in User mode instead of Admin mode ?
> 
> Only solution is to send the command `WebServer 2` either by serial or by MQTT

Can you tell me how i can send these commands trough mqtt?

**Reply by barbudor** [reply]
If you didn't sent 'webserver 1' that may not be the problem.
What have you done to 'disable the http api'?
Depending on what you have done the recipe may be different
May be showing a screenshot could help us understand what you have done

Sending commands through MQTT depends on your environment and tools but quickly you have to send a message on topic 'cmnd/xxxxx/webserver' with payload '2'
How you do that depends on what you have, wheen is you mqtt broker, etc....

**Reply by Barry40** [reply]
> If you didn't sent 'webserver 1' that may not be the problem. What have you done to 'disable the http api'? Depending on what you have done the recipe may be different May be showing a screenshot could help us understand what you have done
> 
> Sending commands through MQTT depends on your environment and tools but quickly you have to send a message on topic 'cmnd/xxxxx/webserver' with payload '2' How you do that depends on what you have, wheen is you mqtt broker, etc....

i did it in the configuration page:
![image](https://user-images.githubusercontent.com/18517382/139395547-e1838317-3478-4a24-ab62-45b98596d044.png)
After doing this, the configuration, console and firmware page is not loading.
It gives me error (ERR_EMPTY_RESPONSE) in chrome.
(This one and some others are still working, the rest (6 devices) are faulty.. 
i have mqtt explorer in windows, but i can't found out how to publish the webserver 2 to the device?

**Reply by sfromis** [reply]
Unticking that "HTTP API enable" means that Tasmota may reject a request going to a subpage when the referer is not the device itself. You should only go to the device home page then, without any path outside of the host name or IP address. With that setting, other pages should be accessed clicking a link in the web UI, not by links saved elsewhere like bookmarks, or copy/paste.

If you have the Tasmota console at the same time, a message like this would be visible:
`HTP: Referer '' denied. Use 'SO128 1' for HTTP API commands. 'Webpassword' is recommended.`

**Reply by Barry40** [reply]
> Unticking that "HTTP API enable" means that Tasmota may reject a request going to a subpage when the referer is not the device itself. You should only go to the device home page then, without any path outside of the host name or IP address. With that setting, other pages should be accessed clicking a link in the web UI, not by links saved elsewhere like bookmarks, or copy/paste.
> 
> If you have the Tasmota console at the same time, a message like this would be visible: `HTP: Referer '' denied. Use 'SO128 1' for HTTP API commands. 'Webpassword' is recommended.`

You're my hero for today!
Hostname didnt load up the pages but the ip does!
i enabled the http api, after that all is working again!
This can be set to solved :) 
Many thanks!

**Reply by sfromis** [reply]
If host name does not work, that is up to your router / DNS setup. This part is not related to disabling http "API".

**Reply by Barry40** [reply]
that is really weird then..
the homepage does open with the hostname, but the configuration part does not.
with the ip it does, i dunno why, but after enabling http api everything works again.

**Reply by sfromis** [reply]
With that "HTTP API" unticked, using the host name for web page access, outside of the Tasmota front page, will only work properly if the host name used to resolve DNS is the same as configured in Tasmota. This is because Tasmota verifies the referer header against the host name and IP address it knows.

**Reply by Barry40** [reply]
> With that "HTTP API" unticked, using the host name for web page access, outside of the Tasmota front page, will only work properly if the host name used to resolve DNS is the same as configured in Tasmota. This is because Tasmota verifies the referer header against the host name and IP address it knows.

Okay :)
Well, everything works again so, thanks all!
